skymp5-server/cpp/lib_espm/espm.cpp

```diff
@@ -1058,3 +1060,62 @@ espm::GMST::Data espm::GMST::GetData(
     compressedFieldsCache);
   return result;
 }
+
+espm::Effects::Effects(const RecordHeader* parent)
+  : parent(parent)
+{
+}
+
+espm::Effects::Data espm::Effects::GetData(
+  CompressedFieldsCache& compressedFieldsCache) const noexcept
+{
+  if (!parent) {
+    return Data();
+  }
+
+  Data result;
+  uint32_t effect = 0;
+
+  espm::RecordHeaderAccess::IterateFields(
+    parent,
+    [&](const char* type, uint32_t size, const char* data) {
+      if (!memcmp(type, "EFID", 4)) {
+        result.effects.emplace_back();
+        result.effects[effect].effectId =
```

Comment:
This is only valid if `effect = 0`. I guess this should have been a map

Reply:
look at new tests